code/hud/hudsquadmsg.cpp

```diff
@@ -832,6 +841,93 @@ bool hud_squadmsg_is_target_order_valid(size_t order, ai_info *aip, bool isWing
 	}
 }
 
+const SCP_string& hud_squadmsg_get_order_name(int command)
+{
+	Assertion(SCP_vector_inbounds(Player_orders, command), "Order does not exist! Get a coder!");
+
+	return Player_orders[command].hud_name;
+}
+
+scripting::api::lua_enum hud_squadmsg_get_order_scripting_enum(int command)
+{
+	Assertion(SCP_vector_inbounds(Player_orders, command), "Order does not exist! Get a coder!");
+
+	switch (command) {
+	case ATTACK_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_ATTACK_TARGET;
+	case DISABLE_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_DISABLE_TARGET;
+	case DISARM_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_DISARM_TARGET;
+	case PROTECT_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_PROTECT_TARGET;
+	case IGNORE_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_IGNORE_TARGET;
+	case FORMATION_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_FORMATION;
+	case COVER_ME_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_COVER_ME;
+	case ENGAGE_ENEMY_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_ENGAGE_ENEMY;
+	case CAPTURE_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_CAPTURE_TARGET;
+	case REARM_REPAIR_ME_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_REARM_REPAIR_ME;
+	case ABORT_REARM_REPAIR_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_ABORT_REARM_REPAIR;
+	case STAY_NEAR_ME_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_STAY_NEAR_ME;
+	case STAY_NEAR_TARGET_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_STAY_NEAR_TARGET;
+	case KEEP_SAFE_DIST_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_KEEP_SAFE_DIST;
+	case DEPART_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_DEPART;
+	case DISABLE_SUBSYSTEM_ITEM:
+		return scripting::api::LE_SQUAD_MESSAGE_DISABLE_SUBSYSTEM;
+	default:
+		return scripting::api::LE_SQUAD_MESSAGE_LUA_AI;
+	}
+}
+
+// Run the order issued hook. When an order is issued we first check if we should override. If so, then we run the hook and return true
+// which will skip the rest of the order code.
+bool hud_squadmsg_run_order_issued_hook(int command, ship* sendingShip, ship* recipientShip, wing* recipientWing, ship* target, ship_subsys* subsys)
+{
+	bool isOverride = false;
+
+	if (scripting::hooks::OnHudCommOrderIssued->isActive()) {
+		object_ship_wing_point_team* recipient = nullptr;
+
+		if (recipientShip != nullptr) {
+			recipient = new object_ship_wing_point_team(recipientShip);
+		} else if (recipientWing != nullptr) {
+			recipient = new object_ship_wing_point_team(recipientWing);
+		}
```

Comment:
This shouldn't be a pointer at all in the first place.
Make it a default / empty oswpt object (without a pointer), and reassign it if either of these criteria is met.
Otherwise, if it stays nullptr, it'll explode when dereferencing for the parameter list anyways.

Reply:
I think I fixed that correctly then.